lib/chef/resource.rb

```diff
@@ -1316,7 +1316,11 @@ class << self
     # life as well.
     @@sorted_descendants = nil
     def self.sorted_descendants
-      @@sorted_descendants ||= descendants.sort_by { |x| x.to_s }
+      # so it turns out Class.to_s just blindly returns the name of the class,
+      # which can be a Symbol in addition to being a String.  as a result, we
+      # may get a Symbol back from Class.to_s, so we call #to_s on that again.
+      # (buggy on at least ruby-2.4.4)
+      @@sorted_descendants ||= descendants.sort_by { |x| x.to_s.to_s }
```

Comment:
This comment is incorrect. We have classes in the tests that define `resource_name` as a symbol and define `self.to_s` to return the resource name. See: https://github.com/chef/chef/pull/4730